### DISCUSSION TITLE
Can you target specific elements where the polyfill will be applied?

### DISCUSSION THREAD

**Discussion by kjetilhartveit** [limitlessloop/flex-gap-polyfill#32] [Q&A] [answered]
As asked in the subject, can you target specific elements where the polyfill will be applied instead of all?

**Comment by gavinmcfarland** [comment, marked as answer]
Not currently. I've thought about this and still wondering what would be the best way to do this. Could you expand a bit on your use case? Is it that the polyfill adds too much weight? Or maybe it interferes with something?

**Reply by kjetilhartveit** [reply]
In a nutshell I just want it to be applied where necessary and use native gap where possible. I haven't tried this package yet, but it sounds promising. However we'd like to avoid unnecessary css/interventions where possible

**Reply by gavinmcfarland** [reply]
Thanks, I think then in your case what might be better is some way to detect if the polyfill is needed rather than only applying the polyfill to certain elements. I say this because if you apply the polyfill to only certain elements, how will the rest of the design cope when `gap` isn't supported? Naturally, the polyfill only applies in cases where the `gap` property is present. In the next release, there is an option to use the polyfill alongside a technique of detecting if flex gap is supported via JavaScript and then the polyfill styles will only be applied in browsers that don't support flex gap.

**Reply by kjetilhartveit** [reply]
Ah good point, conditionally applying the polyfill is a great idea. 

I forgot to mention that the reason I'd like to only apply the polyfill on some elements is that we also use gap with grid so I assumed the polyfill would interfer with grid gaps as well, but perhaps I assumed wrongly? If so, that would be great. Also if that's the case, what would happen if you changed the display property from flex to something else? Specifically we need the polyfill for flexbox on iOS, but they do support gap for grid. 

Sorry for leaving out so much information in my last posts, looks like I had the "Friday" syndrome.

**Reply by gavinmcfarland** [reply]
Hey, no worries at all. Yes so currently the polyfill will only apply the polyfill when used with `display: flex`. Therefore it won't interfere with `grid` but the downside is that both `display: flex` and `gap` need to be inside the same rule.

In the next release there will be support for the following:
- Option to conditionally apply the polyfill by detecting if the browser supports flex gap using JavaScript.
- New feature which will polyfill regardless of whether `display: flex` is inside the same rule as `gap` and doesn't interfere with grid gap, but does increase overall file size.
- Support for TailwindCSS which works by making a special exception for `.flex`, `.m-*` and `.g-` classes.

Oh sorry forgot to answer your last question. If `display` changed from flex to something else dynamically using JavaScript the polyfill would still apply, but the next release will have an option to deal with this by applying a custom property via JavaScript which will prevent the polyfill from being applied.

**Reply by kjetilhartveit** [reply]
Thank you for your replies. I have a good idea of how it works now, looking forward to try it. :)

**Reply by gavinmcfarland** [reply]
Hi. I just wanted to let you know that the polyfill has been rewritten from the ground up and now has a feature where you can restrict it to certain rules.

By default it will polyfill to accommodate every scenario (better experience, but bigger file size). However you can now restrict what rules are polyfilled by passing and array of selectors to match for example:  `flex-gap-polyfill: { only: [/^.m-/] }`.
